### PR TITLE
mmdoc: init at 0.9.1

### PR DESCRIPTION
**Pull request by ryantm**
###### Motivation for this change

Add a CommonMark to HTML renderer that I use for my personal website and maybe we'll use for Nixpkgs/NixOS manuals.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/text/mmdoc/default.nix [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+, libzip
+, ninja
+, meson
+, pkg-config
+}:
+
+let
+
+cmark-gfm' = cmark-gfm.overrideAttrs (oldAttrs: rec {
+  # remove when https://github.com/github/cmark-gfm/pull/248 merged and released
```
Comment (jtojnar) [inline pkgs/tools/text/mmdoc/default.nix:16]: Why not apply this in the main `cmark-gfm` package?

Comment (ryantm) [reply pkgs/tools/text/mmdoc/default.nix]: Sure, done!

**Review by mweinelt — APPROVED** [review]

**Comment by mweinelt** [comment]
Test are failing on aarch64-linux. Did not look at darwin.

**Review by cyplo — APPROVED** [review]
mindforger and mmdoc seem to working on  x86_64-linux

**Comment by ryantm** [comment]
I got it to build on aarch64-linux. It looks like the tests are segfaulting for some reason that I haven't looked into yet, but I tested the outputted binary and it seems to be working fine.

**Pull request merged**